### PR TITLE
Conversion magnetic field to force

### PR DESCRIPTION
**Pull request by annamcraig**

### CODE REVIEW

File: src/mantarray_waveform_analysis/transforms.py
Comment (ejfine) [inline src/mantarray_waveform_analysis/transforms.py]: it might be good here to note here in the docstring that the units of "displacement" are meters

File: src/mantarray_waveform_analysis/pipelines.py
Comment (ejfine) [inline src/mantarray_waveform_analysis/pipelines.py]: Is there a reason you're using the "compressed force" to calculate the metrics instead of regular force?

Comment (ejfine) [reply src/mantarray_waveform_analysis/pipelines.py]: The reason we have compressed data is to be able to pass the raw x/y data stream to the front-end to be able to be animated at a fast frame rate.  But for accuracy of the metrics we typically want to do all our calculations on the non-compressed datastream

File: src/mantarray_waveform_analysis/transforms.py
Comment (ejfine) [reply src/mantarray_waveform_analysis/transforms.py]: I would also add a comment about the date those conversion values were obtained (today), and who they were obtained from (Kevin Grey), so we know if we look back in the future

File: src/mantarray_waveform_analysis/transforms.py
Comment (ejfine) [inline src/mantarray_waveform_analysis/transforms.py]: typically a constant is declared outside of a function (in the main scope of the module) so that line of code isn't executed every single time the function is called.  normally it's not much of a concern, but we do anticipate that the conversion of voltage data to force, peak detection, and calculating metrics will become a bottleneck area in the performance of the Live View of the software that we'll need to optimize. So just trying to keep tiny best practices like this in mind ahead of time

File: tests/test_transforms.py
Comment (ejfine) [inline tests/test_transforms.py]: to help keep the code DRY ("do not repeat yourself") and have a "single source of truth", typically if there's a constant used in both the codes and the tests, then the tests import it from the main code

**Comment by ejfine** [comment]
great work so far. a couple of thoughts/questions. probably best discussed live to make sure we're all on the same page

File: src/mantarray_waveform_analysis/pipelines.py
Comment (ejfine) [inline src/mantarray_waveform_analysis/pipelines.py]: sorry if this wasn't clear earlier---eventually we may need to display to the users metrics calculated either from the magnetic data, displacement data, or force data.  So we should keep this present method like it was before, but we ened to add `get_displacement_data_metrics` and `get_force_data_metrics`

File: tests/test_transforms.py
Comment (ejfine) [inline tests/test_transforms.py]: these values show up in both the code and the tests, so it may be worth refactoring them to a single source of truth as constants somewhere

**Review by ejfine — COMMENTED** [review]
looking good. two quick notes and then I think it'll be ready to show to Kevin.  You can use pip to install this branch in the SDK to start implementing putting the data into the Excel sheet using `get_force_data_metrics` while we're waiting to confirm that this looks good with Kevin before merging into development and officially publishing a new version

**Review by ejfine — DISMISSED** [review]
looks great---pending verification with Kevin it's good to merge

**Review by ejfine — COMMENTED** [review]
thanks for the update.  In `pipelines.py`, can you have the displacement and force metrics be unrounded by default (i.e. use the `rounded=False` kwarg when they call `data_metrics`...I think, if I'm understanding the code correctly)?

File: src/mantarray_waveform_analysis/constants.py
Comment (ejfine) [inline src/mantarray_waveform_analysis/constants.py:34]: did this used to have a comment previously about them being obtained on a certain date from Kevin?

Also, are you waiting for either Kevin or Jason to review the formula? Or have you already tested it with Kevin's sample data and it is good to go?

Comment (annamcraig) [reply src/mantarray_waveform_analysis/constants.py:34]: I was waiting on Kevins review on the formula. I tested it with the sample data and it matched. Kevin said the scaling looks good but is wondering about a subtraction of the baseline, which was for the SDK but he referenced subtraction in the calculation of the amplitude and fwhm. I didn't see anything on the asana card referring to that so waiting on clarification as to what that means.

File: src/mantarray_waveform_analysis/peak_detection.py
Comment (ejfine) [inline src/mantarray_waveform_analysis/peak_detection.py:812]: minor note for future: the time value (x coordinate) will still always be an integer, right?

File: src/mantarray_waveform_analysis/pipelines.py
Comment (ejfine) [inline src/mantarray_waveform_analysis/pipelines.py:99]: note for future: typo `failture`

File: src/mantarray_waveform_analysis/pipelines.py
Comment (ejfine) [inline src/mantarray_waveform_analysis/pipelines.py:258]: note for future: typo `failture`

File: src/mantarray_waveform_analysis/pipelines.py
Comment (ejfine) [inline src/mantarray_waveform_analysis/pipelines.py:362]: note for future: we're not actually using GMR sensors anymore. so maybe something more generic like `get_noise_filtered_sensor_reading` (and generally replacing gmr with `sensor_reading` or something similar)

File: src/mantarray_waveform_analysis/transforms.py
Comment (ejfine) [inline src/mantarray_waveform_analysis/transforms.py:248]: note for future: is this comment relevant here anymore? or do you think having it in the `constants.py` sufficiently covers things?

File: tests/fixtures_utils.py
Comment (ejfine) [inline tests/fixtures_utils.py:188]: note for future: now that we know about the `pytest.approx` method, I think this could probably be removed, and anywhere it is being used replaced with `approx`

**Review by ejfine — APPROVED** [review]
looks good. there's some minor notes for the future to jot down before moving the card over to the "Done" column

**Pull request merged**
